### PR TITLE
11534: Solutions for equations with radicals

### LINKED ISSUE #11534: invert is not working for x**2*y**2 + x**2 - y**2=0, x
```
In [1]: solveset(-y + x/sqrt(-x**2 + 1),x ,S.Reals)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-3da20c9df9b9> in <module>()
----> 1 solveset(-y + x/sqrt(-x**2 + 1),x ,S.Reals)

/root/sympy/sympy/solvers/solveset.pyc in solveset(f, symbol, domain)
    865         return result
    866 
--> 867     return _solveset(f, symbol, domain, _check=True)
    868 
    869 

/root/sympy/sympy/solvers/solveset.pyc in _solveset(f, symbol, domain, _check)
    686                         result += _solve_radical(equation,
    687                                                  symbol,
--> 688                                                  solver)
    689                     elif equation.has(Abs):
    690                         result += _solve_abs(f, symbol, domain)

/root/sympy/sympy/solvers/solveset.pyc in _solve_radical(f, symbol, solveset_solver)
    510                          for g_y in g_y_s])
    511 
--> 512     return FiniteSet(*[s for s in result if checksol(f, symbol, s) is True])
    513 
    514 

TypeError: 'Complement' object is not iterable
```

Original : https://groups.google.com/forum/#!msg/sympy/YtQYNVQGGqY/8dAEcQwkAgAJ

### PR DESCRIPTION
**Pull request by nikoskaragiannakis**
Fixes #11534

In case of equations with radicals, the result is a FiniteSet consisted of
checked solutions of the 'result' calculated inside _solve_radical helper
function.

If the equation to solve does not satisfy the requirements needed in order
to remove the radicals (defined inside sympy.solvers.unrad function),
'cov' is empty and, therefore, the 'result' gets to be a Complement object.
In such a case, the Comlplement needs to be returned, instead of a
FiniteSet object.

For example:

`eq1 = -y + x/sqrt(-x**2 + 1)`
and
`eq2 = -y**2 + x**2/(-x**2 +1)`

should have the same solution, which is

`sol = Complement(FiniteSet(-y/sqrt(y**2 + 1), y/sqrt(y**2 + 1)), FiniteSet(-1, 1))`

### CODE REVIEW

**Comment by asmeurer** [comment]
@sympy/sympy-solvers

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -513,7 +513,9 @@ def _solve_radical(f, symbol, solveset_solver):
         result = Union(*[imageset(Lambda(y, g_y), f_y_sols)
                          for g_y in g_y_s])
 
-    return FiniteSet(*[s for s in result if checksol(f, symbol, s) is True])
+    return (result
```
Comment (Shekharrajak) [inline sympy/solvers/solveset.py]: I tried to do this but  it was returning (ConditionSet \ {-1, 1})

```
In [1]: solveset(-y + x/sqrt(-x**2 + 1),x ,S.Reals)
Out[1]: 
⎧             2  2    2    2    ⎫          
⎨x | x ∊ ℝ ∧ x ⋅y  + x  - y  = 0⎬ \ {-1, 1}
⎩                               ⎭ 
```

Comment (nikoskaragiannakis) [inline sympy/solvers/solveset.py]: I see. The only way to get this output is to define x,y by
```python
x, y = sympy.symbols("x y")
``

If you do
```
x = sympy.Symbol("x", real=True)
y = sympy.Symbol("y", real=True)
```
you'll get the correct answer.

Comment (nikoskaragiannakis) [inline sympy/solvers/solveset.py]: I can check why is this the case in the evening today.

File: sympy/solvers/solveset.py [hunk 2]
```diff
@@ -739,7 +741,7 @@ def solveset(f, symbol=None, domain=S.Complexes):
         A set of values for `symbol` for which `f` is True or is equal to
         zero. An `EmptySet` is returned if `f` is False or nonzero.
         A `ConditionSet` is returned as unsolved object if algorithms
-        to evaluatee complete solution are not yet implemented.
+        to evaluate complete solution are not yet implemented.
```
Comment (Shekharrajak) [inline sympy/solvers/solveset.py]: Changes which are not related to PR/bug, should be avoided.

Comment (nikoskaragiannakis) [inline sympy/solvers/solveset.py]: Ok. Shall I revert this?

Comment (Shekharrajak) [inline sympy/solvers/solveset.py]: Yes

Comment (hargup) [inline sympy/solvers/solveset.py]: > Ok. Shall I revert this?

No. It is generally advised to avoid unrelated changes because review of one change can stall merging of other. Instead, lots of small PRs are recommended, as they are easier to manage, review and merge. This is a really small uncontroversial change, there is no need to "unfix" that change just to create another PR.

Comment (nikoskaragiannakis) [inline sympy/solvers/solveset.py]: I see there is a disagreement here, but I'll be happy to go with whatever you guys decide.

Comment (Shekharrajak) [inline sympy/solvers/solveset.py]: This is just small change, no need of revert. Actually I was just advising for future.

**Comment by smichr** [comment]
> But that means the output of solveset has to change to include the ConditionSet, right?

Yes. We don't want to start letting solveset veer from design. It will not be fun tracking down what is and isn't handled properly. Better to keep it true to design with every modification.

**Comment by nikoskaragiannakis** [comment]
> Yes. We don't want to start letting solveset veer from design. It will not be fun tracking down what is and isn't handled properly. Better to keep it true to design with every modification.

Ok, I'll work on this and update the PR.

**Comment by smichr** [comment]
Thanks for working on this. A future direction is to have a `_solve_radical_real` (I think) because one can request only the real roots from Poly and thus not have to filter out complex solutions. I'm not sure that this is a problem right now or not, though (i.e. I don't know if `checksol` will stumble on determining that a complex solution is not a solution of an expression for which real solutions are being sought).

**Comment by nikoskaragiannakis** [comment]
@smichr It was my pleasure! Shall I update https://github.com/sympy/sympy/wiki/Release-Notes-for-1.1 as well?

**Comment by smichr** [comment]
> If so, under which category

My initial hunch is that this would go under New Features, the new feature being that solutions to radical equations that return a ComplementSet no longer raise an error and solutions to the same which cannot be affirmatively verified to be solutions are returned in a ConditionSet.

File: sympy/solvers/solveset.py (hunk 2 above)
Comment (asmeurer) [inline sympy/solvers/solveset.py]: Fixing spelling mistakes is fine. It's usually better to do it in a separate commit, but if it's near the PR change you don't need a separate PR.
